src/x/cost/enforcer.go

```diff
@@ -122,27 +127,35 @@ func (e *enforcer) checkLimit(cost Cost, limit Limit) error {
 		return nil
 	}
 
-	if e.costMsg == "" {
-		return defaultCostExceededError(cost, limit)
-	}
-	return costExceededError(e.costMsg, cost, limit)
+	return NewCostExceededError(e.costMsg, cost, limit.Threshold)
```

Comment:
nit: I think having the check above be `cost <= limit.Threshold || !limit.Enabled`; would clear up error messages such as `... 6 exceeds limit of 6...` which are a bit questionable imo

Reply:
That looks better, yeah